internal/config/config.go

```diff
@@ -517,6 +533,42 @@ func (c *Config) getFieldIDs(client *jira.Client) (*fields, error) {
 	return &fieldIDs, nil
 }
 
+// getComponents resolves every component set in config against
+// Jira project, and saves these components used by issue-sync.
+func (c *Config) getComponents(proj *jira.Project) ([]*jira.Component, error) {
+	var returnComponents []*jira.Component
+
+	componentsStr := c.cmdConfig.GetString(options.ConfigKeyJiraComponents)
+	if componentsStr != "" {
+		components := strings.Split(componentsStr, ",")
+
+		for i := range components {
+			configComponent := &components[i]
+			found := false
+
+			for j := range proj.Components {
```

Comment:
Same here as above

Reply:
not applicable here, as it'd copy a lot of data for each iteration.